src/agent/src/config.rs

```diff
@@ -589,6 +603,7 @@ mod tests {
                     no_proxy: "",
                     guest_components_rest_api: GuestComponentsFeatures::default(),
                     guest_components_procs: GuestComponentsProcs::default(),
+                    image_registry_auth: "",
```

Comment:
Since the field is under a #[cfg(feature = "guest-pull")], don't you need the same on that code as well?

Reply:
Yea, I think it would be fixed inside #9904 as they are not related to this PR. I will do a rebase after 9904 is merged.